src/ui/actions/session.ts

```diff
@@ -25,77 +33,176 @@ declare global {
   }
 }
 
-// Create a session to use while debugging.
-export async function createSession(store: UIStore, recordingId: string) {
-  addEventListener("Recording.uploadedData", (data: uploadedData) =>
-    store.dispatch(onUploadedData(data))
-  );
-  addEventListener("Recording.awaitingSourcemaps", () =>
-    store.dispatch(actions.setAwaitingSourcemaps(true))
-  );
-  addEventListener("Recording.sessionError", (error: sessionError) =>
-    store.dispatch(
-      setUnexpectedError({
-        ...error,
-        message: "Unexpected session error",
-        content: "The session has closed due to an error. Please refresh the page.",
-        action: "refresh",
-      })
-    )
-  );
-
-  try {
-    ThreadFront.setTest(getTest() || undefined);
-    ThreadFront.recordingId = recordingId;
-
-    if (!isTest()) {
-      tokenManager.addListener(({ token }) => {
-        if (token) {
-          sendMessage("Authentication.setAccessToken", { accessToken: token });
+export function checkRecordingAccessible(
```

Comment:
Nit: Maybe `getAccessibleRecording` instead since it returns either the recording or null. "check" implies a boolean to me